doc/scheduling.md

```diff
@@ -0,0 +1,180 @@
+
+`OP` -- a string comparison operator:
+* **eq** `FIELD` is fully equal to the string specified in `ARG`;
+* **prefix** `FIELD` starts with the string specified in `ARG`;
+* **suffix** `FIELD` ends with the string specified in `ARG`;
+
+`ARG` -- an argument for the operator `OP`.
+
+Example configuration:
+```
+srv_group static { ... }
+srv_group foo_app { ... }
+srv_group foo_beta { ... }
+srv_group bar_app { ... }
+
+sched_http_rules {
+	match static   uri       prefix  "/static" ;
+	match static   uri       suffix  ".php";
+	match static   host      prefix  "static.";
+	match static   host      suffix  "tempesta-tech.com";
+	match foo_beta host      eq      "beta-foo.example.com" backup=foo_app;
+	match foo_app  host      eq      "foo.example.com";
+	match bar_app  hdr_conn  eq      "keep-alive";
+	match bar_app  hdr_host  prefix  "bar.";
+	match bar_app  hdr_host  suffix  "natsys-lab.com";
+	match bar_app  hdr_host  eq      "bar.natsys-lab.com";
+	match bar_app  hdr_raw   prefix  "X-Custom-Bar-Hdr: ";
+}
+```
+
+**HTTP Scheduler** matches request to rules in _top-down order_. If requests
+fits some rule, rule is applied and further matching is skipped. User can provide
+default match rule that matches any request. If no default rule provided and
+[default server group]() is defined, default match rule to group `default` will
+be added implicitly to the end of the list. The syntax is as follows:
+```
+match <SRV_GROUP> * * * ;
+```
+
+
+## Scheduling Requests inside Server Groups
+
+**Server Schedulers** are bound to exact server groups, see
+[Setting scheduler]() section in [Server Group]() description.
+
+User can override default server scheduler for all groups using `sched`
+configuration option:
+```
+sched <SCHED_NAME>;
+```
+
+
+### Round-Robin Scheduler
+
+**TODO:** update to Ratio and extended weights.
+
+Rotates all servers in a group in round-robin manner, parallel connections to
+the same server are also rotated in the round-robin manner, so requests are
+distributed to all the servers in a group in a fair way.
+
+Scheduler name for use in configuration file: `round-robin`.
+
+
+### Hash Scheduler
+
+Chooses server to schedule request to by hashed key value. Key is built using
+_uri_, _request method_ and _host header_ of the request. In most cases load will
+be distributed among all the servers in group, but situations when single server
+pulls all the load are also possible. Although it is quite improbable, such
+condition is quite stable: it cannot be fixed by adding/removing servers and
+restarting Tempesta FW.
+
+Hash scheduler have an advantages though. It schedules requests of same resources
+to the same servers. That can reduce load of backend servers.
+
+Scheduler name for use in configuration file: `hash`.
+
+
+## Scheduling HTTP sessions to the same server
+
+None of the schedulers is responsible for forwarding requests from the same HTTP
+session to the same server. [Sticky Sessions]() option should be used instead.
```

Comment:
This is Wiki, so it should be friendly to beginners, so please describe when a user should choose the scheduler, when it's important and useful. Also it's good to say in RR section that RR is very fast and basically provides good enough load distribution. Hash section already has "Hash scheduler have an advantages though..."